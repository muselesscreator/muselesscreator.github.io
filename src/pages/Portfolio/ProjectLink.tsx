type ProjectLinkProps = {
  IconComponent: React.ComponentType,
  href?: string,
  label?: string,
};

const ProjectLink = ({
  IconComponent,
  href,
  label,
}: ProjectLinkProps) => (
  (href && label) && (
    <p>
      <IconComponent />: 
      <a href={href} target="_blank" className="project-link" rel="noopener noreferrer">
        {label}
      </a>
    </p>
  )
);

export default ProjectLink;

