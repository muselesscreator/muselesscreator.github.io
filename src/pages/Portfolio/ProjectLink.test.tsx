import 'react';
import {
  vi,
  describe,
  test,
  expect,
} from 'vitest';
import { shallow } from '@muselesscreator/react-shallow-snapshot';
import ProjectLink from './ProjectLink';

vi.mock('react-icons/fa', () => ({ FaEye: 'FaEyeIcon' }));

const props = {
  IconComponent: () => <div>IconComponent</div>,
  href: 'test-href',
  label: 'Test Label',
};
describe('ProjectLink', () => {
  test('snapshot', () => {
    const el = shallow(<ProjectLink  {...props} />);
    expect(el.snapshot).toMatchSnapshot();
    const { children } = el.instance.findByType('p')[0];
    const expected = props.IconComponent();
    expect(children[0].matches(expected)).toBe(true);
    const link = el.instance.findByType('a')[0];
    expect(link.props.href).toBe(props.href);
    expect(link.children[0].el).toBe(props.label);
  });
});
