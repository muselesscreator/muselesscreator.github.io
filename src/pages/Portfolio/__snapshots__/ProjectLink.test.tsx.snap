// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectLink > snapshot 1`] = `
<p>
  <div>
    IconComponent
  </div>
  :
  <a
    className="project-link"
    href="test-href"
    rel="noopener noreferrer"
    target="_blank"
  >
    Test Label
  </a>
</p>
`;
